### DISCUSSION TITLE
nvim 0.6.1 稳定配置文件及插件包

### DISCUSSION THREAD

**Discussion by AGou-ops** [AGou-ops/dotfiles#6] [General]
nvim 0.6.1  stable config and plugins compressed package.

- `nvim0.6.1_config_stable.tar.gz`：配置文件目录，`~/.config/nvim`
- `vimPlugged_0.6.1_stable.tar.gz`： 插件文件目录，`~/.vim/plugged`
- `local_lspServers_0.6.1_stable.tar.gz`：lsp servers文件目录，`~/.local/share/nvim`

<hr /><em>This discussion was created from the release <a href='https://github.com/AGou-ops/dotfiles/releases/tag/nvim0.6.1-stable'>nvim 0.6.1 稳定配置文件及插件包</a>.</em>

**Comment by frog-game** [comment]
能否弄一个一键安装的bash批处理安装呢，

**Reply by AGou-ops** [reply]
这个脚本没啥好写的吧，把压缩包解压到指定的目录就可以了.
比如`tar xf nvim0.6.1_config_stable.tar.gz -C ~/.config/nvim`就完事了.

**Reply by frog-game** [reply]
![image](https://user-images.githubusercontent.com/31720616/166192832-d8b5bbad-54dd-47a9-b875-91d2a04c2bef.png)
楼主请教下用这个lazygit 控件的时候，上下选择 当前选项高亮不了你有遇到过吗

**Reply by AGou-ops** [reply]
没遇到过，只是在neovim中用的时候，高亮颜色会很浅.

**Reply by frog-game** [reply]
好吧，看来只能去lazygit官网他源码找找看看能不能改过来了~

**Reply by frog-game** [reply]
![image](https://user-images.githubusercontent.com/31720616/166695017-c017a66f-44b9-4b50-9c0e-04f68cf663cf.png)
这个找到原因了
改这个配置加上这个就好了
![image](https://user-images.githubusercontent.com/31720616/166695356-8960ab93-b611-447c-841d-209469b957cb.png)

**Comment by frog-game** [comment]
up主 请教下vimspector能调试多虚拟机的lua代码吗，比如skynet这种

**Reply by AGou-ops** [reply]
不知道，`vimspector`我个人只是拿来debug单台物理机的代码，用的也很基础，没有多虚拟机的用途，你去[vimspector官方仓库](https://github.com/puremourning/vimspector)文档里面看看吧，README写的很详细.

**Reply by frog-game** [reply]
好的，我去看看

**Comment by liyongzheng666** [comment]
UP主，这个是需要先按照视频那个安装完成之后，再把这几个压缩包进行覆盖嘛？

**Reply by AGou-ops** [reply]
不必完全按照视频上面的来，当然你按照视频上的来也可以，直接解压覆盖是没问题的。
也可以安装好nvim0.6.1版本之后，将这几个压缩包解压到指定目录就可以了.

**Reply by liyongzheng666** [reply]
好的，我再尝试一下。另外一个问题是，如果我想把这个覆盖到我的云服务器也是可以的对吧？小白一个，搜资料的话基本没有人这样干，往UP抽空指点一二哈哈哈。

**Reply by AGou-ops** [reply]
可以的，服务器上和本地基本没什么区别，就是渲染上（比如字体）可能比不上本地，显示的效果会差一些.

**Reply by liyongzheng666** [reply]
老哥,我现在还有一个疑惑就是neovim_lua这个文件夹如何下载下来，我拉取到自己的仓库来了，也还是不可以，我也不知道这属于是啥问题，能不能指教一下。

**Reply by AGou-ops** [reply]
你直接终端运行下面的命令就可以了（克隆仓库直接拷贝过去）：

```bash
git clone https://github.com/AGou-ops/dotfiles.git
cd dotfiles
cp -a neovim-with-lua/* ~/.config/nvim
```

或者拉取指定`nvim`分支，这个分支只有nvim的配置文件：

```bash
git clone -b nvim https://github.com/AGou-ops/dotfiles.git
cp -a dotfile/* ~/.config/nvim
```

俩都可以，你自己选择吧.

**Reply by AGou-ops** [reply]
如果clone慢的话，直接页面上手动点下载，然后拷贝到指定目录，也是一样的.

**Reply by liyongzheng666** [reply]
终于弄好了，一直都是我的网络问题，无法拉取到内容。感谢UP的悉心指点，点赞！！

**Comment by frog-game** [comment]
lua 使用gd 和gr命令貌似不生效~

**Reply by AGou-ops** [reply]
你检查下你的lua lsp装好了没.

**Reply by frog-game** [reply]
![image](https://user-images.githubusercontent.com/31720616/166655268-4bb53ec7-ed4f-4461-b7b7-917b8094d9b1.png)
确定装好了~你看，感觉是不是按键冲突了，或者什么原因，我按gd gr等等命令都没有反应，但是如果直接命令模式下输入lua vim.lsp.buf.declaration()

**Reply by frog-game** [reply]
lua vim.lsp.buf.definition()  Lspsaga rename<cr>这些命令都能够有反应~~

**Comment by frog-game** [comment]
![image](https://user-images.githubusercontent.com/31720616/166655605-a1b3c17b-c082-4d2f-a2da-bbde9fc50141.png)

**Reply by frog-game** [reply]
up主你看~~
如果按gr 就没反应

**Reply by AGou-ops** [reply]
那你再检查一下`lsp.lua`这个文件的第`53`行的`servers `这个变量有没有把lua的lsp加进去（我个人没加，因为很少用到这些东西，我只拿来补全😅）.

https://github.com/AGou-ops/dotfiles/blob/a5c09dbf094ece03cab565006840ffb1add1814c/neovim-with-lua/lua/AGou/lsp.lua#L52-L60

**Reply by AGou-ops** [reply]
你应该用的是`sumneko_lua`吧，直接加在后面就可以了，把没用的lsp server也可以从这里删去.

**Reply by frog-game** [reply]
嗯呢，我用的是sumeko_lua 我试试哈，辛苦up主

**Reply by frog-game** [reply]
后面我加sumneko_lua试过，也不行~~感觉好奇怪，我看从64行开始up主有对sumeko _lua的设置呢，而且我还把luajit 改成了lua 5.3也不起作用，智能提示到是有，改名什么的快捷键就是不起作用~~

**Reply by AGou-ops** [reply]
欸，不对，我对lua的lsp有专门的配置，就在下面，不在我上面说的哪里加（加了也没问题）.
我这里刚试了试用快捷键是可以的。
![2022-05-04 18-04-00 2022-05-04 18_05_14](https://user-images.githubusercontent.com/57939102/166661870-46912bfc-6741-499f-a1b0-3cde536e03a9.gif)

**Reply by AGou-ops** [reply]
但是会出现两次，应该是我哪里配置重复了，有空我再检查一下配置.

**Reply by frog-game** [reply]
我这直接按了就没反应~~

**Reply by AGou-ops** [reply]
把cmd那行删掉就正常只出现一次了.

**Reply by frog-game** [reply]
1，我试试~~ 3q  up主

**Reply by frog-game** [reply]
![image](https://user-images.githubusercontent.com/31720616/166663280-b4e21c2d-e9c8-4f69-bd4c-8d7650b4a12c.png)
是这个cmd是吗

**Reply by AGou-ops** [reply]
嗯，我这里是的.

**Reply by AGou-ops** [reply]
或者把runtime全删了也可以.

**Reply by frog-game** [reply]
![image](https://user-images.githubusercontent.com/31720616/166663974-1f3ce78f-9adf-4776-98b1-f512bedc1347.png)
还是不行，都搞成这样了

**Reply by AGou-ops** [reply]
保留一个，保留vim那一行，或者runtime那一段.
改了之后如果还是不行那就真奇怪了，在我这里是正常的.😅

**Reply by frog-game** [reply]
![image](https://user-images.githubusercontent.com/31720616/166664860-5e9ea222-e0cb-49a6-a45d-475866b65c2e.png)

改完后变成这样了

**Reply by AGou-ops** [reply]
这又是什么奇怪的报错，按理说补全可以用，然后加上`on_attach`函数就能正常使用的了.
要不你就重新安装下lua的lsp，把lua和luajit版本升级一下再试试看.

**Reply by frog-game** [reply]
好的

**Reply by frog-game** [reply]
![image](https://user-images.githubusercontent.com/31720616/166694710-cc0ccb4b-f4c1-4a3b-a56e-6651c5b35a5c.png)
实在找不到原因了，up主，有时间帮找找吧，我现在暂时用的coc.vim 实现的上图的查找所有引用~~ LSP试了好多法子，还是GG了

**Reply by AGou-ops** [reply]
这。。我没你那边的环境也没法给你找啊，同样的配置我这边是可以的，要不你就找找其他人的配置改改看.

**Comment by frog-game** [comment]
好的。。



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: AGou-ops ***@***.***&gt;
发送时间: 2022年5月4日 18:44
收件人: AGou-ops/dotfiles ***@***.***&gt;
抄送: 505384662 ***@***.***&gt;, Comment ***@***.***&gt;
主题: 回复：[AGou-ops/dotfiles] nvim 0.6.1 稳定配置文件及插件包 (Discussion #6)





 
这又是什么奇怪的报错，按理说补全可以用，然后加上on_attach函数就能正常使用的了.
 要不你就重新安装下lua的lsp，把lua和luajit版本升级一下再试试看.

**Comment by frog-game** [comment]
好吧



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: AGou-ops ***@***.***&gt;
发送时间: 2022年5月4日 23:07
收件人: AGou-ops/dotfiles ***@***.***&gt;
抄送: 505384662 ***@***.***&gt;, Comment ***@***.***&gt;
主题: 回复：[AGou-ops/dotfiles] nvim 0.6.1 稳定配置文件及插件包 (Discussion #6)





 
这。。我没你那边的环境也没法给你找啊，同样的配置我这边是可以的，要不你就找找其他人的配置改改看.

**Comment by frog-game** [comment]
![image](https://user-images.githubusercontent.com/31720616/166900422-14a3abfd-6748-4c4e-8af2-026af2dce034.png)
up主你那个session 退出的时候能下载进来 按,sl命令然后还是以这个窗口还原打开吗

**Reply by frog-game** [reply]
up主你那个session 退出的时候能下次进来 按,sl命令然后还是以这个窗口还原打开吗

**Reply by AGou-ops** [reply]
能还原buffer，排版不能完全还原，我很少用这玩意.

**Reply by frog-game** [reply]
那有好的插件能推荐，还原排版吗~· 让我去试试~

**Reply by AGou-ops** [reply]
不知道，我之前用过两个，都没法完美还原排版，你自己找找吧.

**Reply by frog-game** [reply]
好吧

**Comment by frog-game** [comment]
![image](https://user-images.githubusercontent.com/31720616/167295833-f3217e73-4df5-4231-931b-ee61cc50c6f5.png)
up主还打扰你一下，lsp我找到什么原因导致的按键失效问题了，请教下你这个跳转到定义是那个按键呢，还是说lsp就不支持函数跳转。。

**Reply by AGou-ops** [reply]
`gd`快捷键就是函数定义跳转啊.

**Reply by frog-game** [reply]
试了，跳不过去，。头大

**Reply by AGou-ops** [reply]
`:message`看看有什么错误提示吗？

**Reply by frog-game** [reply]
![image](https://user-images.githubusercontent.com/31720616/167297765-ffd6fc38-b461-45ea-b1a6-5965a7c765be.png)

啥都没，看着挺正常的

**Reply by AGou-ops** [reply]
那就不知道了.🥲

**Reply by frog-game** [reply]
好吧~~

**Comment by frog-game** [comment]
https://user-images.githubusercontent.com/31720616/167300386-ed58da27-9a0c-443f-8cb5-5da623f99dcb.mp4

可以了 up主 我在sumneko_lua 官网把他最新的源码编译后替换就好了。。。。

**Reply by AGou-ops** [reply]
🎉

**Comment by frog-game** [comment]
![image](https://user-images.githubusercontent.com/31720616/167459211-f68dbb2e-0944-42e5-ad89-3fa6729ff669.png)
up主你这里有个字段 prompt_prefix = "> ", 这个“”里面有两个字符会导致我试用vim 的yw  ，p这样的复制粘贴命令粘贴不到里面去，不知道为什么，盲猜这个有没有可能是telescope 插件的bug  后来我给改成了这样
![image](https://user-images.githubusercontent.com/31720616/167459566-973d61f9-011c-41d4-9371-dc3e6b45489a.png)
虽然少了个提示符号，但是却能粘贴，真的好奇怪

https://user-images.githubusercontent.com/31720616/167459980-aad540e5-4b13-41cf-9032-d35052ab9925.mp4

**Reply by AGou-ops** [reply]
不知道，可能是bug吧，这个只是个提示输入的前缀，我这里是没问题的.

**Reply by frog-game** [reply]
好吧
